gtsfm/data_association/point3d_initializer.py

```diff
@@ -55,6 +56,54 @@ class TriangulationParam(Enum):
     RANSAC_TOPK_BASELINES = 3  # deterministically choose hypotheses with largest estimate baseline
 
 
+class TriangulationOptions(NamedTuple):
+    """Options for triangulation solver.
+
+    Refs:
+      - https://github.com/colmap/colmap/blob/dev/src/optim/ransac.h
+      - http://www.cse.psu.edu/~rtc12/CSE486/lecture15.pdf
+
+    Args:
+        reproj_error_threshold: the maximum reprojection error allowed.
+        mode: triangulation mode, which dictates whether or not to use robust estimation.
+        min_inlier_ratio: a priori assumed minimum probability that a point is an inlier.
+        confidence: desired confidence that at least one hypothesis is outlier free.
+        dyn_num_hypotheses_multiplier: multiplication factor for dynamically computed hyptheses based on confidence.
+        min_num_hypotheses: minimum number of hypotheses.
+        mas_num_hypotheses: masimum number of hypotheses.
+    """
+
+    reproj_error_threshold: float
+    mode: TriangulationParam
```

Comment:
I feel like TriangulationParam is not a good name for this enum. It means the same thing as `TriangulationOptions` but the enum is about what type of ransac to run. Maybe RansacMode or something is better.

Reply:
I'm ok with it currently. I see what you mean Akshay, but I think `RansacMode.NO_RANSAC` would be sort of weird.

Reply:
Maybe `SamplingMode` is better?

Reply:
I like `SamplingMode`

Reply:
or `TriangulationSamplingMode`?

Reply:
I updated this, marking as resolved